### ISSUE TITLE
getGitRemote function is failing because the GitHub API call no longer works

### ISSUE THREAD

**Issue by hoylen** [digitalmethodsinitiative/dmi-tcat#195]
The `getGitRemote` function in _capture/common/functions.php_ is failing and putting a lot of warning messages in the Apache _/var/log/apache2/error.log_.

Looks like whatever it is using _curl_ to retrieve has changed.

The code is:
```php
function getGitRemote($compare_local_commit = '', $branch = 'master') {
    if (!function_exists('curl_init')) return false;
    if (!defined('REPOSITORY_URL')) {
        $repository_url = 'https://api.github.com/repos/digitalmethodsinitiative/dmi-tcat/commits';
    } else {
        $repository_url = REPOSITORY_URL;
    }
    $repository_url .= '?sha=' . $branch;
    $ch = curl_init($repository_url);
    curl_setopt($ch, CURLOPT_USERAGENT, 'DMI-TCAT GIT remote version checker (contact us at https://github.com/digitalmethodsinitiative/dmi-tcat)');
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    $output = curl_exec($ch);
    curl_close($ch);
    $data = json_decode($output, true);
    $commit = $mesg = $url = $date = null;
    $required = false;
    foreach ($data as $ent) {
        if ($commit === null) {
            $commit = $ent['sha'];
            $mesg = $ent['commit']['message'];
            $url = $ent['html_url'];
            $date = $ent['commit']['committer']['date'];
        }
        if ($ent['sha'] == $compare_local_commit) { break; }
        if (isset($ent['commit']['message'])) {
            if (stripos($ent['commit']['message'], '[required]') !== FALSE) {
                $required = true; break;
            }
        }
    }
   ...
```

The `$output` gets set to this JSON encoded string value:
```{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}```

That causes `$ent` (in the _foreach_ loop) to become the string "Not Found" instead of an array. And the Apache error log then contains many warnings as it tries to index into that string using strings as keys.

```
PHP Warning:  Illegal string offset 'sha' in ...
PHP Warning:  Illegal string offset 'commit' in ...
PHP Warning:  Illegal string offset 'message' in ...
PHP Warning:  Illegal string offset 'html_url' in ...
PHP Warning:  Illegal string offset 'commit' in ...
PHP Warning:  Illegal string offset 'committer' in ...
PHP Warning:  Illegal string offset 'date' in ...
PHP Warning:  Illegal string offset 'sha' in ...
```

**Comment by ErikBorra** [comment]
Hi @hoylen,

I've checked our error logs and we do not get this warnings. When trying the URL https://api.github.com/repos/digitalmethodsinitiative/dmi-tcat/commits we get appropriate JSON returned. Furthermore, I ran the code in the function as a standalone CLI file and that runs fine without any errors. Dumping $data shows the JSON transformed into an associative error.

Maybe you could do some further digging at your end, as I cannot reproduce this issue?

Best,

Erik

**Comment by dentoir** [comment]
I think it only occurs when you have checked out something different than master? If memory serves me, the /commits function (in an unauthenticated API request, like we do it) actually only exists for the default branch of the repository (master in this case)?

**Comment by hoylen** [comment]
I suspected it might be because I installed it from a different repository and branch.

I see it now. The install script should also update the value of `REPOSITORY_URL`. I'll fix get that fixed tomorrow.

Thanks.

**Comment by hoylen** [comment]
I've fixed it and added it to pull request #194, since those were changes to the install script that made it easier to install from different repositories and/or branches.

I've changed the `REPOSITORY_URL` constant to be the real repository URL (instead of the API URL). The code converts that into the corresponding API URL (by replacing "https://github.com/" with "https://api.github.com/repos/", and stripping any ".git" from the end and appending "/commits"). Besides matching the name of the constant, users are more familiar with the repository URL that the API URL.

I spotted the code that invokes `git` to discover the branch name. Maybe the repository URL can be discovered the same way, so no configuration is necessary. A future enhancement...

I've tested it, and it seems to ~~work~~ not generate errors when the branch is not called "master".
